File: app/src/main/java/com/qalliance/treetracker/TreeTracker/fragments/ExitFragment.java
package com.qalliance.treetracker.TreeTracker.fragments;


import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.view.HapticFeedbackConstants;
import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.ViewGroup;
import android.widget.Button;

import com.qalliance.treetracker.TreeTracker.activities.MainActivity;
import com.qalliance.treetracker.TreeTracker.R;
import com.qalliance.treetracker.TreeTracker.utilities.ValueHelper;

public class ExitFragment extends Fragment implements OnClickListener {
	
	private Fragment fragment;
	private Bundle bundle;
	private FragmentTransaction fragmentTransaction;
	private SharedPreferences mSharedPreferences;

	public ExitFragment() {
		//some overrides and settings go here
	}

	@Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        
    }
	
	@Override
	public void onResume() {
		super.onResume();
	}
	
    	 
	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup container,
	        Bundle savedInstanceState) {
		
	    View v = inflater.inflate(R.layout.fragment_exit, container, false);
	    
	    mSharedPreferences = getActivity().getSharedPreferences(
	      	      "com.qalliance.treetracker", Context.MODE_PRIVATE);

	    Button yesBtn = (Button) v.findViewById(R.id.fragment_exit_yes);
	    yesBtn.setOnClickListener(ExitFragment.this);

	    Button noBtn = (Button) v.findViewById(R.id.fragment_exit_no);
	    noBtn.setOnClickListener(ExitFragment.this);
	    
	    return v;
	}

	public void onClick(View v) {
		
		
		v.performHapticFeedback(HapticFeedbackConstants.VIRTUAL_KEY, HapticFeedbackConstants.FLAG_IGNORE_GLOBAL_SETTING);
		
		switch (v.getId()) {
			case R.id.fragment_exit_yes:

				MainActivity.syncDataFromExitScreen = true;
				
				fragment = new DataFragment();
				bundle = getActivity().getIntent().getExtras();
				fragment.setArguments(bundle);
				
				fragmentTransaction = getActivity().getSupportFragmentManager()
						.beginTransaction();
				fragmentTransaction.replace(R.id.container_fragment, fragment).addToBackStack(ValueHelper.DATA_FRAGMENT).commit();
				
				FragmentManager manager = getActivity().getSupportFragmentManager();
				FragmentManager.BackStackEntry first = manager.getBackStackEntryAt(0);
				manager.popBackStack(first.getId(), FragmentManager.POP_BACK_STACK_INCLUSIVE);
				
				break;		
			case R.id.fragment_exit_no:
				getActivity().finish();
				break;					
		}

		
	}


}

File: app/src/main/java/org/greenstand/android/TreeTracker/fragments/ExitFragment.kt
package org.greenstand.android.TreeTracker.fragments


import android.content.Context
import android.content.SharedPreferences
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentTransaction
import android.view.HapticFeedbackConstants
import android.view.LayoutInflater
import android.view.View
import android.view.View.OnClickListener
import android.view.ViewGroup
import android.widget.Button

import org.greenstand.android.TreeTracker.activities.MainActivity
import org.greenstand.android.TreeTracker.R
import org.greenstand.android.TreeTracker.utilities.ValueHelper

class ExitFragment : Fragment(), OnClickListener {

    private var fragment: Fragment? = null
    private var bundle: Bundle? = null
    private var fragmentTransaction: FragmentTransaction? = null
    private var mSharedPreferences: SharedPreferences? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

    }

    override fun onResume() {
        super.onResume()
    }


    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {

        val v = inflater!!.inflate(R.layout.fragment_exit, container, false)

        mSharedPreferences = activity.getSharedPreferences(
                "org.greenstand.android", Context.MODE_PRIVATE)

        val yesBtn = v.findViewById(R.id.fragment_exit_yes) as Button
        yesBtn.setOnClickListener(this@ExitFragment)

        val noBtn = v.findViewById(R.id.fragment_exit_no) as Button
        noBtn.setOnClickListener(this@ExitFragment)

        return v
    }

    override fun onClick(v: View) {


        v.performHapticFeedback(HapticFeedbackConstants.VIRTUAL_KEY, HapticFeedbackConstants.FLAG_IGNORE_GLOBAL_SETTING)

        when (v.id) {
            R.id.fragment_exit_yes -> {

                MainActivity.syncDataFromExitScreen = true

                fragment = DataFragment()
                bundle = activity.intent.extras
                fragment!!.arguments = bundle

                fragmentTransaction = activity.supportFragmentManager
                        .beginTransaction()
                fragmentTransaction!!.replace(R.id.container_fragment, fragment).addToBackStack(ValueHelper.DATA_FRAGMENT).commit()

                val manager = activity.supportFragmentManager
                val first = manager.getBackStackEntryAt(0)
                manager.popBackStack(first.id, FragmentManager.POP_BACK_STACK_INCLUSIVE)
            }
            R.id.fragment_exit_no -> activity.finish()
        }


    }


}//some overrides and settings go here
